core/domain/user_query_domain.py

```diff
@@ -124,7 +135,12 @@ def validate(self):
                 % self.sent_email_model_id)
 
     @classmethod
-    def create_default(cls, query_id, query_params, submitter_id):
+    def create_default(
+        cls: Type[T],
+        query_id: str,
+        query_params: UserQueryParams,
+        submitter_id: str
+    ) -> T:
```

Comment:
Why are generics (`T`) needed here?

Reply:
It returns UserQuery the same instance of the class

Reply:
@vojtechjelinek Here self-typing is implemented as we used in our models. Thanks!

Reply:
But I'm not sure why.

Reply:
it returns the same instance of the class hence added , besides if I remove it error is thrown in the checks

Reply:
Still not sure why it is needed `-> UserQuery` should work too, right?

Reply:
that was giving an error and it was based on this reference https://stackoverflow.com/questions/44640479/mypy-annotation-for-classmethod-returning-instance